e2e/test/scenarios/metrics/metrics-creation.cy.spec.ts

```diff
@@ -0,0 +1,251 @@
+      saveMetric();
+      runQuery();
+      verifyScalarValue("18,760");
+    });
+
+    it("should create a metric for a model", () => {
+      cy.visit("/");
+      startNewMetric();
+      popover().findByText("Models").click();
+      popover().findByText("Orders Model").click();
+      addAggregation("Count of rows");
+      saveMetric();
+      runQuery();
+      verifyScalarValue("18,760");
+    });
+
+    it("should create a metric for another metric", () => {
+      createOrderCountMetric();
+      cy.visit("/");
+      startNewMetric();
+      popover().findByText("Metrics").click();
+      popover().findByText("Orders, Count").click();
+      saveMetric();
+      runQuery();
+      verifyScalarValue("18,760");
+    });
+  });
+
+  describe("joins", () => {
+    it("should join a table", () => {
+      cy.visit("/");
+      startNewMetric();
+      popover().findByText("Raw Data").click();
+      popover().findByText("Products").click();
+      cy.button("Join data").click();
+      popover().findByText("Orders").click();
+      popover().findByText("ID").click();
+      popover().findByText("Product ID").click();
+      startNewFilter();
+      popover().within(() => {
+        cy.findByText("User").click();
+        cy.findByText("State").click();
+        cy.findByText("CA").click();
+        cy.button("Add filter").click();
+      });
+      addAggregation("Count of rows");
+      saveMetric();
+      runQuery();
+      verifyScalarValue("613");
+    });
+  });
+
+  describe("breakouts", () => {
+    it("should create a timeseries metric", () => {
+      cy.visit("/");
+      startNewMetric();
+      popover().findByText("Raw Data").click();
+      popover().findByText("Orders").click();
+      addAggregation("Sum of ...", "Total");
+      addBreakout("Created At");
+      saveMetric();
+      runQuery();
+      verifyLineChart("Created At", "Sum of Total");
+    });
+
+    it("should create a geo metric", () => {
+      cy.visit("/");
+      startNewMetric();
+      popover().findByText("Raw Data").click();
+      popover().findByText("People").click();
+      addAggregation("Count of rows");
+      addBreakout("Latitude");
+      addBreakout("Longitude");
+      saveMetric();
+      runQuery();
+      verifyPinMap();
+    });
+  });
+
+  describe("aggregations", () => {
+    it("should create a metric with a custom aggregation expression based on 1 metric", () => {
+      createOrderCountMetric();
+      cy.visit("/");
+      startNewMetric();
+      popover().findByText("Metrics").click();
+      popover().findByText("Orders, Count").click();
+      getNotebookStep("summarize").findByText("Orders, Count").click();
+      enterCustomColumnDetails({ formula: "[Orders, Count] / 2 ", name: "" });
+      popover().button("Update").click();
+      saveMetric();
+      runQuery();
+      verifyScalarValue("9,380");
+    });
+
+    it("should create a metric with a custom aggregation expression based on 2 metrics", () => {
```

Comment:
This one fails atm because of a BE bug

Reply:
Nit: Can you please reference that issue as the reproduction in the test title?

Reply:
Fixed